gen3-integration-tests/tests/test_gen3_workflows.py

```diff
@@ -0,0 +1,147 @@
+import json
+
+import pytest
+from services.gen3workflows import Gen3Workflow
+from utils import TEST_DATA_PATH_OBJECT
+
+
+def is_service_missing(service_name: str):
+    manifest_path = f"{TEST_DATA_PATH_OBJECT}/configuration/manifest.json"
+    manifest_data = json.loads(manifest_path)
+    return any(service not in manifest_data["versions"] for service in service_name)
+
+
+@pytest.mark.skipif(
+    is_service_missing("funnel"),
+    reason="funnel service is not running on this environment",
+)
+@pytest.mark.skipif(
+    is_service_missing("gen3-workflow"),
+    reason="gen3-workflow is not running on this environment",
+)
+@pytest.mark.gen3_workflow
+class TestGen3Workflow(object):
+    gen3_workflow = Gen3Workflow()
+    valid_user = "main_account"
+    invalid_user = "dummy_one"
+    bucket_name = None
+    folder_name = "integration-tests"
+    file_name = "test-input.txt"
+
+    ######################## Test /storage/info endpoint ########################
+    def test_get_storage_info_with_valid_token(self):
+        """Test GET /storage/info with a valid access token."""
+        response = self.gen3_workflow.get_storage_info(
+            user=self.valid_user, expected_status=200
+        )
+        assert isinstance(response, dict), "Expected a valid JSON response"
+        assert (
+            "bucket" in response
+        ), "Expected the response to contain the `S3 bucket` details"
+        self.bucket_name = response["bucket"]
+
+    def test_get_storage_info_without_token(self):
+        """Test GET /storage/info without an access token."""
+        self.gen3_workflow.get_storage_info(user=None, expected_status=401)
+
+    ######################## Test /s3/ endpoint ########################
+
+    @pytest.mark.skipif(
+        not bucket_name,
+        reason="Skipping this test, since a valid s3 bucket name couldn't be found ",
+    )
+    def test_any_user_cannot_get_s3_file_with_unsigned_request(self):
+        """Unsigned requests should receive 401 even if the user is authorized to get data."""
+        self.gen3_workflow.get_bucket_object_with_unsigned_request(
```

Comment:
this function is only used in this test, i'd say it'd make sense to just run the function contents here directly

Reply:
I understand why you’d suggest this, and I agree that it would simplify things. However, I preferred placing the logic in the Gen3Workflow class to abstract the concept of users and Gen3Auth from the tests, allowing Gen3Workflow to handle it within the service layer.